### DISCUSSION TITLE
TP4 - Ejercicio 4 - División lenta II

### DISCUSSION THREAD

**Discussion by vlzJuan** [INGCOM-UNRN-P1/cursada-2023#147] [Preguntas y respuestas] [answered]
Tengo una consulta para aclaracion sobre la consigna:

### Ejercicio 4 - División lenta II

Implementar una única función que obtenga el cociente y resto de dos números enteros.

¿La idea es que la funcion retorne ambas cosas, las imprima? ¿Deberia seleccionar cual retornar? ¿Que es exactamente lo que debería dar como salida?

**Comment by martinvilu** [comment]
La idea de las funciones es que retornen el valor siempre (salvo las excepciones que vimos en clase)

> ¿Que es exactamente lo que debería dar como salida?

Ambos valores simultaneamente.

**Reply by vlzJuan** [reply]
La cosa es que el cociente y el resto son dos variables distintas. ¿Como podría hacer que una sola funcion retorne las dos cosas al mismo tiempo? 

Por ejemplo, en Octave se puede hacer algo como

```MATLAB
[a,b] = function cosa(variable):
%cosas
endfunction
```

¿Hay alguna forma de hacer esto en C? Si no solo se me ocurre imprimir para dar los dos resultados simultaneamente

**Reply by martinvilu** [reply, marked as answer]
Nope, pero sí es posible “retornar” varias cosas simultáneamente con punteros.

```C
int division_lenta(int dividendo, int divisor, int* cociente, int* resto);
```
Fíjate que la función retorna igual algo, es un buen lugar para retornar sobre los posibles fallos de la función, en especial porque hay dos punteros y porque la función no puede dar un resultado cuando el divisor es = 0.

Con esto, las postcondiciones son más importantes que antes, ya que deben registrar que lo apuntado por los punteros cambia.

**Reply by vlzJuan** [reply]
Esto es lo que necesitaba, gracias!
